### ISSUE TITLE
How to concatenate two (or more) options?

### ISSUE THREAD

**Issue by fernandodev** [hawx/clive#1]
Hi, Nice library!

How to concatenate two (or more) options?

Example:

```
git commit -am
-a -> `all`
-m -> `commit message`
```

**Comment by hawx** [comment]
It should "just work". For example, try the following with `-am "My message"`

``` ruby
class CLI < Clive
  opt :a, :all
  opt :m, :message, args: '<message>'
end

result = CLI.run 
puts "all: #{result[:all]}"
puts "message: #{result[:message]}"
```

It's probably worth noting that if you try `-ma "Hey"`, instead, it will not work since Clive wants the final short name to be the one that takes the argument.

**Comment by fernandodev** [comment]
Yeah! It works fine, but when I try to put into a command scope it doesn't work.

``` ruby
class CLI < Clive
  command :commit do
    opt :a, :all do |all|
      @all = all
    end
    opt :m, :message, args: '<message>' do |message|
      @message = message
    end

    action do 
      puts "all: #{@all}, message: #{@message}"
    end
  end
end

result = CLI.run 
```

**Comment by hawx** [comment]
Oh wow, nice spot. Seems I wrote that part of the parser assuming there would only ever be uncombined arguments. It should be fixed now, try v1.2.1. Thanks.

**Comment by fernandodev** [comment]
Thank you very much @hawx!!!
